src/computation/StreamedComputation.cpp

```diff
@@ -71,16 +71,29 @@ void StreamedComputation::performComputation(std::shared_ptr<wrench::ActionExecu
             // WRENCH_INFO("Chunk: %.2lf bytes / %.2lf flops", num_bytes, num_flops);
             // Start the computation asynchronously
             simgrid::s4u::ExecPtr exec = simgrid::s4u::this_actor::exec_init(num_flops);
-            exec->start();
-            double exec_start_time = exec->get_start_time();
-            // Read data from the file
-            read_start_time = wrench::Simulation::getCurrentSimulatedDate();
-            fs.second->getStorageService()->readFile(fs.first, fs.second, num_bytes);
-            read_end_time = wrench::Simulation::getCurrentSimulatedDate();
-            // Wait for the computation to be done
-            exec->wait();
+            double exec_start_time = 0.0;
+            double exec_end_time = 0.0;
+            if(this->prefetching_on){
+                exec->start();
+                exec_start_time = exec->get_start_time();
+                // Read data from the file
+                read_start_time = wrench::Simulation::getCurrentSimulatedDate();
+                fs.second->getStorageService()->readFile(fs.first, fs.second, num_bytes);
+                read_end_time = wrench::Simulation::getCurrentSimulatedDate();
+                // Wait for the computation to be done
+                exec->wait();
+                exec_end_time = exec->get_finish_time();
+            }
+            else {
```

Comment:
I see what your point is here, but I think in this case it might be better to set an integer variable which sets the number of prefetching blocks, instead of a simple bool.

That way, one would have a closer representation of what is possible in XRootD.

Reply:
Ok I'll have a look at a better implenentation, thanks!

Reply:
@HerrHorizontal I've adapted the implementation. Please have a look :)